goss-testing/tests/ncn/goss-switch-ping-status.yaml

```diff
@@ -0,0 +1,41 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+#{{ $scripts := .Env.GOSS_BASE | printf "%s/scripts" }}
+#{{ $logrun := $scripts | printf "%s/log_run.sh" }}
+command:
+    {{ $testlabel := "switch-ping-status" }}
+    {{$testlabel}}:
+        title: Verify that the network switch has external access
+        meta:
+            desc: If this test fails, validate the network switch configurations. See troubleshooting/known_issues/check_system_management_monitoring_tools.md in the CSM documentation for remediation steps.
+            sev: 0
+        exec: |-
+            #Set pipefail so that if any commands in the command pipeline fail, the test will fail
+            set -o pipefail
+            "{{$logrun}}" --no-args -l "{{$testlabel}}" \
+            canu test --csm 1.2 --json --username admin --password ${SW_ADMIN_PASSWORD}
+        stdout:
+            - "!/ping.*FAIL/"
```

Comment:
This means that it will pass if no lines of output contain that pattern. It also needs to make sure that the ping test actually passed, though, not just that it did not fail.

Reply:
This comment was marked as resolved but the issue still exists.